airflow/ti_deps/deps/base_ti_dep.py

```diff
@@ -81,6 +81,10 @@ def get_dep_statuses(self, ti, session, dep_context):
         :param dep_context: the context for which this dependency should be evaluated for
         :type dep_context: DepContext
         """
+        from airflow.ti_deps.dep_context import DepContext
```

Comment:
The motivation is probably clearer in the [original commit](https://github.com/Avocarrot/incubator-airflow/commit/4653fd8b92fb5507a547318117e5d188b1f7e072) before the squash. While it's possible to make `dep_context` optional without a local import, it requires further refactoring and this PR is already big enough so I opted for the less invasive option. But feel free to refactor it or revert the previous signature (and update the dozens previously broken tests that were passing `None`) if you feel strongly about it.